agent/agenttest/client.go

```diff
@@ -264,9 +228,13 @@ func (f *FakeAgentAPI) GetServiceBanner(context.Context, *agentproto.GetServiceB
 	return agentsdk.ProtoFromServiceBanner(sb), nil
 }
 
-func (*FakeAgentAPI) UpdateStats(context.Context, *agentproto.UpdateStatsRequest) (*agentproto.UpdateStatsResponse, error) {
-	// TODO implement me
-	panic("implement me")
+func (f *FakeAgentAPI) UpdateStats(ctx context.Context, req *agentproto.UpdateStatsRequest) (*agentproto.UpdateStatsResponse, error) {
+	f.logger.Debug(ctx, "update stats called", slog.F("req", req))
+	// empty request is sent to get the interval; but our tests don't want empty stats requests
+	if req.Stats != nil {
+		f.statsCh <- req.Stats
+	}
+	return &agentproto.UpdateStatsResponse{ReportInterval: durationpb.New(statsInterval)}, nil
```

Comment:
Does it mean that tests are time dependent on real`statsInterval`? If so, it is possible to refactor them to use `clock`?

Reply:
yes, they are, and yes, it's possible.  But, I think that's outside the scope of this PR.  I've just copied the hardcoded value from before into a `const`.

Reply:
Oh, I assumed that it would be a low-hanging fruit to refactor, but I'm good with leaving for further.